Packages/Tracking/Core/Runtime/Scripts/HandPoses/HandPoseDetector.cs

```diff
@@ -0,0 +1,373 @@
+        /// </summary>
+        public enum TypeOfDirectionCheck
+        {
+            OBJECT = 0,
+            WORLD = 1,
+            CAMERALOCAL = 2
+        };
+
+        /// <summary>
+        /// Adding the ablility of enum style direction selection in the Inspector.
+        /// </summary>
+        public enum AxisToFace { Back, Down, Forward, Left, Right, Up, Zero };
+        private static readonly Vector3[] vectorAxes = new Vector3[]
+        {
+            Vector3.back,
+            Vector3.down,
+            Vector3.forward,
+            Vector3.left,
+            Vector3.right,
+            Vector3.up,
+            Vector3.zero
+        };
+        public Vector3 GetAxis(AxisToFace axis)
+        {
+            return vectorAxes[(int)axis];
+        }
+
+        /// <summary>
+        /// Add more here to put constraints on when the pose is detected. 
+        /// E.g. if the user needs to point at an object with their index finger for the pose to be considered detected"
+        /// </summary>
+        [SerializeField]
+        public List<FingerDirection> BoneDirectionTargets;
+
+        /// <summary>
+        /// Holds information about different fingers, bones and their directional targets.
+        /// </summary>
+        [Serializable]
+        public struct FingerDirection
+        {
+            public bool enabled;
+            public TypeOfDirectionCheck typeOfDirectionCheck;
+            public bool isPalmDirection;
+            public Transform poseTarget;
+            public AxisToFace axisToFace;
+            public Finger.FingerType fingerTypeForPoint;
+            public Bone.BoneType boneForPoint;
+            public float rotationThreshold;
+        }
+        #endregion
+
+        public void CreateDefaultFingerDirection()
+        {
+            FingerDirection fingerDirection= new FingerDirection();
+            fingerDirection.typeOfDirectionCheck = TypeOfDirectionCheck.CAMERALOCAL;
+            fingerDirection.isPalmDirection = false;
+            fingerDirection.poseTarget = null;
+            fingerDirection.axisToFace = AxisToFace.Forward;
+            fingerDirection.fingerTypeForPoint = Finger.FingerType.TYPE_INDEX;
+            fingerDirection.boneForPoint = Bone.BoneType.TYPE_DISTAL;
+            fingerDirection.rotationThreshold = 15;
+            fingerDirection.enabled = true;
+
+            BoneDirectionTargets.Add(fingerDirection);
+        }
+        public void RemoveDefaultFingerDirection(int index)
+        {
+            BoneDirectionTargets.RemoveAt(index);
+        }
+
+
+
+        public static event Action<HandPoseScriptableObject> PoseHasBeenDetected;
+        public static event Action PoseHasNotBeenDetected;
+
+        private void Start()
+        {
+            //PoseHasBeenDetected += PoseDetected;
+            //PoseHasNotBeenDetected += PoseNotDetected;
+            if(_leapProvider == null)
+            {
+                _leapProvider = FindObjectOfType<LeapProvider>();
```

Comment:
if this used Hands.Provider, it would find the most suitable provider rather than just the first one in the scene :)